asset-transfer-basic/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,246 @@
+
+        //Get the asset details by assetID
+        await readAssetByID(contract);
+
+        //Update an asset which does not exist.
+        await updateAsset(contract)
+
+    } finally {
+        gateway.close();
+        client.close();
+    }
+
+}
+
+main().catch(error => console.error('******** FAILED to run the application:', error));
+
+async function newGrpcConnection(): Promise<grpc.Client> {
+    const tlsRootCert = await fs.readFile(tlsCertPath);
+    const tlsCredentials = grpc.credentials.createSsl(tlsRootCert);
+    const GrpcClient = grpc.makeGenericClientConstructor({}, '');
+    return new GrpcClient(peerEndpoint, tlsCredentials, {
+        'grpc.ssl_target_name_override': 'peer0.org1.example.com',
+    });
+}
+
+async function newIdentity(): Promise<Identity> {
+    const credentials = await fs.readFile(certPath);
+    return { mspId, credentials };
+}
+
+async function newSigner(): Promise<Signer> {
+    const privateKeyPem = await fs.readFile(keyPath);
+    const privateKey = crypto.createPrivateKey(privateKeyPem);
+    return signers.newPrivateKeySigner(privateKey);
+}
+
+async function initLedger(contract:Contract):Promise<void> {
+    // This type of transaction would only be run once by an application the first time it was started after it
+    // deployed the first time. Any updates to the chaincode deployed later would likely not need to run
+    // an "init" type function.
+
+    console.log(
+        '\n--> Submit Transaction: InitLedger, function creates the initial set of assets on the ledger',
+    );
+
+    // Submit a transaction, blocking until the transaction has been committed on the ledger
+    await contract.submitTransaction('InitLedger');
+
+    console.log('*** Result: committed');
+
+}
+
+async function getAllAssets(contract:Contract):Promise<void> {
+    console.log(
+        '\n--> Evaluate Transaction: GetAllAssets, function returns all the current assets on the ledger',
+    );
+    const result = await contract.evaluateTransaction('GetAllAssets');
+    console.log(`*** Result: ${Buffer.from(result).toString()}`);
+}
+
+async function createAsset(contract:Contract):Promise<void> {
+    console.log(
+        '\n--> Submit Transaction: CreateAsset, creates new asset with ID, color, owner, size, and appraisedValue arguments',
+    );
+    await contract.submitTransaction(
+        'CreateAsset',
+        'asset13',
+        'yellow',
+        '5',
+        'Tom',
+        '1300',
+    );
+    console.log('*** Result: committed');
+}
+
+async function updateAssetAsync(contract:Contract):Promise<void> {
+
+    // Submit transaction asynchronously, blocking until the transaction has been sent to the orderer, and allowing
+    // this thread to process the chaincode response (e.g. update a UI) without waiting for the commit notification
+
+    const commit = await contract.submitAsync('UpdateAsset', {
+        arguments: ['asset1', 'blue', '5', 'Tomoko', '400'],
+    });
```

Comment:
Does this (or any other) transaction that we submit actually return a value? If so, it might be good to print the return value (`commit.getResult()`) immediately after the `submitAsync()` call to demonstrate that we can do work with the result after successfully sending to the orderer but before we wait for the commit

Reply:
I don't see any submit  transaction returning value in the chaincode .